### DISCUSSION TITLE
Mouse clickable buttons

### DISCUSSION THREAD

**Discussion by blacktop** [charmbracelet/bubbletea#76] [Q&A] [answered]
Do you have an examples of making certain text a clickable button with keyboard or mouse? For example the dialogBox in the lipgloss example?  How would you make those buttons clickable?

Thank you!

**Comment by meowgorithm** [comment, marked as answer]
Hi! Mouse integration is still pretty naïve: you’ll need to map click coordinates to the cell coordinates of items in your UI and test hit detection accordingly.

We'd like to ultimately provide a higher-level API for this, but in the meantime check out [Gembro](https://git.sr.ht/~rafael/gembro), a mouse-driven Gemini Browser built on Bubble Tea. It's doing an excellent at job handling mouse clicks under the hood.

**Reply by blacktop** [reply]
Ah, that's cool.  I didn't know you could move issues to discussions.  Thank you for the tip and thank you for creating awesome Go pkgs ❤️

**Comment by Adjective-Object** [comment]
Just dropping this here for anyone else who stumbles on this discussion: https://github.com/lrstanley/bubblezone

After reading this discussion, I spent an afternoon essentially reimplementing half of bubblezone within the bubbletea codebase. here's hoping this saves someone else the trouble in the future 😅

**Reply by blacktop** [reply]
so this functionality is being added to bubbletea proper now?

**Reply by Adjective-Object** [reply]
I don't think so, I'm just leaving the link here as a breadcrumb for the next person who finds this thread.
